### PR TITLE
[AIRFLOW-4771] Initialization of hook in the GCP operator constructors

### PR DESCRIPTION
**Pull request by turbaszek**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [ ] Passes `flake8`

### CODE REVIEW

**Comment by turbaszek** [comment]
@mik-laj

File: airflow/contrib/operators/gcp_compute_operator.py [hunk 1]
```diff
@@ -40,24 +40,27 @@ class GceBaseOperator(BaseOperator):
     """
 
     @apply_defaults
-    def __init__(self,
```
Comment (mik-laj) [inline airflow/contrib/operators/gcp_compute_operator.py]: This looks like unrelated changes. Can you revert it?

File: airflow/contrib/operators/gcp_compute_operator.py [hunk 2]
```diff
@@ -40,24 +40,27 @@ class GceBaseOperator(BaseOperator):
...
         self.gcp_conn_id = gcp_conn_id
         self.api_version = api_version
         self._validate_inputs()
-        self._hook = GceHook(gcp_conn_id=self.gcp_conn_id, api_version=self.api_version)
+        self.gcp_conn_id = gcp_conn_id
         super().__init__(*args, **kwargs)
 
     def _validate_inputs(self):
-        if self.project_id == '':
+        if self.project_id == "":
```
Comment (mik-laj) [inline airflow/contrib/operators/gcp_compute_operator.py]: This looks like unrelated changes. Can you revert it?

File: airflow/contrib/operators/gcp_compute_operator.py [hunk 3]
```diff
@@ -93,26 +96,43 @@ class GceInstanceStartOperator(GceBaseOperator):
...
-                 gcp_conn_id='google_cloud_default',
-                 api_version='v1',
-                 *args, **kwargs):
+    def __init__(
+        self,
+        zone,
+        resource_id,
+        project_id=None,
+        gcp_conn_id="google_cloud_default",
+        api_version="v1",
```
Comment (mik-laj) [inline airflow/contrib/operators/gcp_compute_operator.py]: This looks like unrelated changes. Can you revert it?

File: airflow/contrib/operators/gcp_compute_operator.py [hunk 4]
```diff
@@ -140,31 +160,46 @@ class GceInstanceStopOperator(GceBaseOperator):
...
+        hook = GceHook(gcp_conn_id=self.gcp_conn_id, api_version=self.api_version)
+        hook.stop_instance(
+            zone=self.zone, resource_id=self.resource_id, project_id=self.project_id
+        )
 
 
-SET_MACHINE_TYPE_VALIDATION_SPECIFICATION = [
-    dict(name="machineType", regexp="^.+$"),
-]
+SET_MACHINE_TYPE_VALIDATION_SPECIFICATION = [dict(name="machineType", regexp="^.+$")]
```
Comment (mik-laj) [inline airflow/contrib/operators/gcp_compute_operator.py]: This looks like unrelated changes. Can you revert it?

File: airflow/contrib/operators/gcp_compute_operator.py [hunk 5]
```diff
@@ -197,69 +232,104 @@ class GceSetMachineTypeOperator(GceBaseOperator):
...
+        validate_body=True,
+        *args,
+        **kwargs
+    ):
         self.body = body
         self._field_validator = None
         if validate_body:
             self._field_validator = GcpBodyFieldValidator(
-                SET_MACHINE_TYPE_VALIDATION_SPECIFICATION, api_version=api_version)
+                SET_MACHINE_TYPE_VALIDATION_SPECIFICATION, api_version=api_version
```
Comment (mik-laj) [inline airflow/contrib/operators/gcp_compute_operator.py]: This looks like unrelated changes. Can you revert it?

File: airflow/contrib/operators/gcp_compute_operator.py [hunk 6]
```diff
@@ -197,69 +232,104 @@ class GceSetMachineTypeOperator(GceBaseOperator):
...
+            )
         super().__init__(
-            project_id=project_id, zone=zone, resource_id=resource_id,
-            gcp_conn_id=gcp_conn_id, api_version=api_version, *args, **kwargs)
+            project_id=project_id,
+            zone=zone,
```
Comment (mik-laj) [inline airflow/contrib/operators/gcp_compute_operator.py]: This looks like unrelated changes. Can you revert it?

File: airflow/contrib/operators/gcp_compute_operator.py [hunk 7]
```diff
@@ -197,69 +232,104 @@ class GceSetMachineTypeOperator(GceBaseOperator):
...
-            dict(name="onHostMaintenance", optional=True),
-            dict(name="automaticRestart", optional=True),
-            dict(name="preemptible", optional=True),
-            dict(name="nodeAffinitites", optional=True),  # not validating deeper
-        ]),
-        dict(name="labels", optional=True),
-        dict(name="guestAccelerators", optional=True),  # not validating deeper
-        dict(name="minCpuPlatform", optional=True),
-    ]),
+    dict(
```
Comment (mik-laj) [inline airflow/contrib/operators/gcp_compute_operator.py]: This looks like unrelated changes. Can you revert it?

File: airflow/contrib/operators/gcp_compute_operator.py [hunk 8]
```diff
@@ -276,7 +346,7 @@ def execute(self, context):
     "properties.networkInterfaces.accessConfigs.kind",
     "properties.networkInterfaces.name",
     "properties.metadata.kind",
-    "selfLink"
+    "selfLink",
```
Comment (mik-laj) [inline airflow/contrib/operators/gcp_compute_operator.py]: This looks like unrelated changes. Can you revert it?

**Review by mik-laj — CHANGES_REQUESTED** [review]
Can you delete unrelated changes from this PR? I was not able to fully check the content of this PR. I feel lost.

**Comment by turbaszek** [comment]
Sorry for the inconvenience 😔 I removed all not related changes.

**Comment by turbaszek** [comment]
Will be done after moving GCP to core.

**Comment by turbaszek** [comment]
Done via #5893
